frb_dart/lib/src/env/io.dart

```diff
@@ -0,0 +1,31 @@
+// see https://github.com/flutter/flutter/issues/55870
+
+import 'dart:io';
+
+const truthyValues = ['true', '1', 'yes'];
+
+/// ✅ correctly read e.g. export JSON=true && ...
+/// ❌ fail at reading e.g. dart --define=SILICON=true ...
+/// ❌ fail at reading e.g. dart -DSILICON=true ...
+bool? readEnvFromPlatform(String envVarName) =>
+    (Platform.environment[envVarName] != null &&
+            Platform.environment[envVarName]!.isNotEmpty)
+        ? truthyValues.contains(Platform.environment[envVarName]!.toLowerCase())
+        : null;
+
+/// ✅ correctly read e.g. dart --define=SILICON=true ...
+/// ✅ correctly read e.g. dart -DSILICON=true ...
+/// ❌ fail at reading e.g. export JSON=true && ...
+bool? readEnvFromPrimitive(String envVarName) =>
+    bool.hasEnvironment(envVarName) ? bool.fromEnvironment(envVarName) : null;
+
+bool readBoolEnv(String envVarName, {bool defaultValue = false}) =>
+    readEnvFromPrimitive(envVarName) ??
+    readEnvFromPlatform(envVarName) ??
+    defaultValue;
+
+final isWeb = readBoolEnv('dart.library.html');
+final int? portEnv = Platform.environment['PORT'] != null
+    ? int.parse(Platform.environment['PORT']!)
+    : null;
+final silicon = readBoolEnv('SILICON');
```

Comment:
nit: IIRC, "portEnv" and "silicon" variable is not used in frb_dart package, but only used in frb_example or serve.dart? Thus, not sure whether we should put it *here*

Reply:
- `silicon` is used to conditionally open dylib in ffi.
- `portEnv` is used for the server in dart lib.

Reply:
Yes, that's why maybe we should move it to corresponding places instead of frb_dart package

Reply:
but doesn't that mean duplicating them in most `frb_example` packages ?

Reply:
> but doesn't that mean duplicating them in most frb_example packages ?

Then we may need a package like `frb_example/example_utility` and let all examples import it. Another way is just have current duplication given they are examples.